### PR TITLE
[4.0] Cleanup the View files com_admin

### PR DESCRIPTION
**Pull request by roland-d**
### Summary of Changes
This is a first step in cleaning up the view files and adhering to coding standards as well as making the views more usable by the IDE by removing the magic getter.

### Testing Instructions
1. Go to System -> System Information
2. Check that the page still works
3. Click on the `Download as text`
4. Check that the file is generated
5. Click on the `Download as JSON`
6. Check that the file is generated
7. Click on Help in the left menu
8. Click on Joomla! Help
9. Check that the page still shows

### Documentation Changes Required
None

### CODE REVIEW

**Comment by pe7er** [comment]
I have tested this item :white_check_mark: successfully on e06d1933f4937600fffccf106232c8e55cf0dd50<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25034">issues.joomla.org/tracker/joomla-cms/25034</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e06d1933f4937600fffccf106232c8e55cf0dd50<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25034">issues.joomla.org/tracker/joomla-cms/25034</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25034">issues.joomla.org/tracker/joomla-cms/25034</a>.</sub>

**Comment by Hackwar** [comment]
Pleasebe aware that these changes are PHP7.2.

**Comment by Hackwar** [comment]
But  otherwise love it. Thanks for this work.

**Comment by roland-d** [comment]
@Hackwar Joomla 4 has a minimum requirement of PHP 7.2. Other than that, yes, people should realize this is indeed the case.

**Comment by Hackwar** [comment]
Not yet... Right now the minimum version is still declared 7.0. I completely agree with all the changes and also with going to 7.2, but it would have to be announced first (and changed in code) that we are switching to 7.2. See my PRs related to that, too.

**Comment by roland-d** [comment]
Pinging @wilsonge 😕

**Comment by wilsonge** [comment]
Holding on this until the official announcement comes in production minutes for this if it's requiring 7.2 (and we can update composer files etc)

File: administrator/components/com_admin/View/Sysinfo/HtmlView.php [hunk 1]
```diff
@@ -31,7 +33,7 @@ class HtmlView extends BaseHtmlView
 	 * @var    array
 	 * @since  1.6
 	 */
-	protected $php_settings = array();
+	protected $phpSettings = array();
```
Comment (Quy) [inline administrator/components/com_admin/View/Sysinfo/HtmlView.php]:
```suggestion
	protected $phpSettings = [];
```

File: administrator/components/com_admin/View/Sysinfo/JsonView.php [hunk 2]
```diff
@@ -58,9 +62,9 @@ public function display($tpl = null)
...
 	 * @since   3.5
 	 */
-	protected function getLayoutData()
+	protected function getLayoutData(): array
 	{
-		/* @var \Joomla\Component\Admin\Administrator\Model\SysInfoModel $model */
+		/** @var SysInfoModel $model */
 		$model = $this->getModel();
 
 		return array(
```
Comment (Quy) [inline administrator/components/com_admin/View/Sysinfo/JsonView.php]: Use short syntax.

File: administrator/components/com_admin/View/Sysinfo/TextView.php [hunk 3]
```diff
@@ -75,9 +79,9 @@ public function display($tpl = null)
...
 	 * @since   3.5
 	 */
-	protected function getLayoutData()
+	protected function getLayoutData(): array
 	{
-		/* @var \Joomla\Component\Admin\Administrator\Model\SysInfoModel $model */
+		/** @var SysInfoModel $model */
 		$model = $this->getModel();
 
 		return array(
```
Comment (Quy) [inline administrator/components/com_admin/View/Sysinfo/TextView.php]: Use short syntax.

File: administrator/components/com_admin/View/Sysinfo/TextView.php [hunk 4]
```diff
@@ -119,7 +123,7 @@ protected function getLayoutData()
 	 *
 	 * @since   3.5
 	 */
-	protected function renderSection($sectionName, $sectionData, $level = 0)
+	protected function renderSection(string $sectionName, array $sectionData, int $level = 0): string
 	{
 		$lines = array();
```
Comment (Quy) [inline administrator/components/com_admin/View/Sysinfo/TextView.php]:
```suggestion
		$lines = [];
```

File: administrator/components/com_associations/View/Association/HtmlView.php [hunk 5]
```diff
@@ -28,40 +36,176 @@
...
 
 	/**
-	 * The model state
+	 * The type item
 	 *
-	 * @var    object
+	 * @var    string
+	 * @since  3.7.0
+	 */
+	protected $itemtype;
```
Comment (Quy) [inline administrator/components/com_associations/View/Association/HtmlView.php]: Above `public $itemType = null;`. They are similar except for `T`. Change?

File: administrator/components/com_associations/View/Association/HtmlView.php [hunk 6]
```diff
@@ -228,6 +378,7 @@ protected function addToolbar()
 		ToolbarHelper::cancel('association.cancel', 'JTOOLBAR_CLOSE');
 		ToolbarHelper::help('JHELP_COMPONENTS_ASSOCIATIONS_EDIT');
 
-		\JHtmlSidebar::setAction('index.php?option=com_associations');
+		\JHtmlSidebar::setAction('index.php?option=com_associationsss');
+//		HTMLHelper::_('sidebar.setAction', 'index.php?option=com_associations');
```
Comment (Quy) [inline administrator/components/com_associations/View/Association/HtmlView.php]: Remove comment?

File: administrator/components/com_associations/tmpl/association/edit.php [hunk 7]
```diff
@@ -13,18 +13,20 @@
...
 HTMLHelper::_('behavior.formvalidator');
 HTMLHelper::_('behavior.keepalive');
 HTMLHelper::_('jquery.framework');
 
 HTMLHelper::_('script', 'com_associations/sidebyside.js', ['version' => 'auto', 'relative' => true]);
 HTMLHelper::_('stylesheet', 'com_associations/sidebyside.css', ['version' => 'auto', 'relative' => true]);
 
-$options = array(
+$options = [
 			'layout'   => $this->app->input->get('layout', '', 'string'),
```
Comment (Quy) [inline administrator/components/com_associations/tmpl/association/edit.php]: Reduce indentation.

File: administrator/components/com_associations/View/Association/HtmlView.php [hunk 8]
```diff
@@ -228,6 +378,7 @@ protected function addToolbar()
 		ToolbarHelper::cancel('association.cancel', 'JTOOLBAR_CLOSE');
 		ToolbarHelper::help('JHELP_COMPONENTS_ASSOCIATIONS_EDIT');
 
-		\JHtmlSidebar::setAction('index.php?option=com_associations');
+		\JHtmlSidebar::setAction('index.php?option=com_associationsss');
```
Comment (brianteeman) [inline administrator/components/com_associations/View/Association/HtmlView.php]:
```suggestion
		\JHtmlSidebar::setAction('index.php?option=com_associations');
```

File: administrator/components/com_associations/View/Association/HtmlView.php (hunk 5 above)
Comment (roland-d) [reply administrator/components/com_associations/View/Association/HtmlView.php]: This file should not have been included.

File: administrator/components/com_associations/View/Association/HtmlView.php (hunk 6 above)
Comment (roland-d) [reply administrator/components/com_associations/View/Association/HtmlView.php]: This file should not have been included.

File: administrator/components/com_associations/tmpl/association/edit.php (hunk 7 above)
Comment (roland-d) [reply administrator/components/com_associations/tmpl/association/edit.php]: This file should not have been included.

File: administrator/components/com_associations/View/Association/HtmlView.php (hunk 8 above)
Comment (roland-d) [reply administrator/components/com_associations/View/Association/HtmlView.php]: This file should not have been included.

**Comment by Quy** [comment]
Is PR #13415 started to do what you are doing with these PRs? If yes, should it be closed due to being possibly outdated?

**Comment by roland-d** [comment]
@Quy It is also a change to standardize on our CamelCase codestyle but in a completely different place. So it is a valid PR in it's own right. Yes, it is a very old PR, I don't know if the owner wants to update or if it is still needed, that depends on the conflicts.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**